internal/k8s/reconciler/listener.go

```diff
@@ -195,7 +195,7 @@ func (l *K8sListener) canBind(ctx context.Context, ref gwv1alpha2.ParentReferenc
 			return false, nil
 		}
 
-		if !route.MatchesHostname(l.listener.Hostname) {
```

Comment:
Last comment. I believe pretty much all of this file can get deleted too at this point since you're pulling/setting state from `Gateway` at this point, but you'll have to update the initializers if you drop this. Feel free to do so now or in a follow-up.

Reply:
Handled in https://github.com/hashicorp/consul-api-gateway/pull/337